pkgs/development/python-modules/pyarrow/default.nix

```diff
@@ -63,12 +61,27 @@ buildPythonPackage rec {
   PYARROW_WITH_FLIGHT = zero_or_one _arrow-cpp.enableFlight;
   PYARROW_WITH_HDFS = zero_or_one true;
   PYARROW_WITH_PARQUET = zero_or_one true;
-  PYARROW_WITH_PLASMA = zero_or_one (!stdenv.isDarwin);
+  PYARROW_WITH_PARQUET_ENCRYPTION = zero_or_one true;
+  # Plasma is deprecated since arrow 10.0.0
+  PYARROW_WITH_PLASMA = zero_or_one false;
   PYARROW_WITH_S3 = zero_or_one _arrow-cpp.enableS3;
-
-  PYARROW_CMAKE_OPTIONS = [
-    "-DCMAKE_INSTALL_RPATH=${ARROW_HOME}/lib"
-  ];
+  PYARROW_WITH_GCS = zero_or_one _arrow-cpp.enableGcs;
+  # do this manually to avoid having to patch pyarrow, to handle pyarrow's
+  # assumption of permissions of bundled arrow-cpp headers
+  PYARROW_BUNDLE_ARROW_CPP_HEADERS = zero_or_one false;
+
+  PYARROW_CMAKE_OPTIONS =
+    let
+      components = [
+        "Arrow"
+        "ArrowDataset"
+        "ArrowFlight"
+        "Parquet"
+      ];
+    in
+    [
+      "-DCMAKE_INSTALL_RPATH=${ARROW_HOME}/lib"
+    ] ++ map (c: "-D${c}_DIR=${ARROW_HOME}/lib/cmake/${c}") components;
```

Comment:
Why is this needed?

Reply:
pyarrow doesn't seem to be able to find arrow-cpp's cmake config files without this. I don't know exactly why this is now a requirement.